packages/fogbender/src/renderUnreadBadge.tsx

```diff
@@ -18,31 +18,39 @@ export function renderUnreadBadge(
     attach(el.shadowRoot);
   }
 
-  const x = render(() => {
-    const [unreadCounter, setUnreadCounter] = createSignal(events.unreadCount);
+  const root = el.shadowRoot;
+  if (!root) throw new Error("No shadow root");
 
-    events.on("fogbender.unreadCount", e => {
-      setUnreadCounter(e.unreadCount);
-    });
+  render(<UnreadButton events={events} openWindow={openWindow} />, root);
 
-    return (
-      <button
-        onClick={openWindow}
-        class={tw`min-w-1rem min-h-1rem bg-brand-red-500 text-2xs hidden items-center justify-center rounded-full font-bold leading-none text-white`}
-        style={{
-          display: !unreadCounter() || unreadCounter() === 0 ? "none" : "flex",
-        }}
-      >
-        {unreadCounter() === -1 ? (
-          <span class={tw`text-xs`}>@</span>
-        ) : (
-          <span class={tw`px-1`}>{unreadCounter()}</span>
-        )}
-      </button>
-    );
-  }, el.shadowRoot!);
   return () => {
-    x();
+    render(null, root);
```

Comment:
oh wow, can't belive this is a real API 😬 https://github.com/preactjs/preact/issues/53#issuecomment-184868295